components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java

```diff
@@ -1309,6 +1346,251 @@ public void addApplicationWithTemplateIdAndTemplateVersionData(String templateId
                 REGISTRY_SYSTEM_USERNAME);
     }
 
+    @Test(groups = "certificate", priority = 1)
+    public void testAddApplicationWithCertificate() throws CertificateMgtException {
+
+        ServiceProvider inputSP = new ServiceProvider();
+        inputSP.setApplicationName(APPLICATION_NAME_1);
+        inputSP.setCertificateContent(certificate.getCertificateContent());
+        addApplicationConfigurations(inputSP);
+
+        reset(applicationCertificateManagementService);
+        doReturn(0).when(applicationCertificateManagementService).addCertificate(any(), anyString());
+        doReturn(certificate).when(applicationCertificateManagementService).getCertificateByName(any(), anyString());
+        try {
+            appIdWithCert = applicationManagementService.createApplication(inputSP, SUPER_TENANT_DOMAIN_NAME,
+                    REGISTRY_SYSTEM_USERNAME);
+        } catch (Exception e) {
+            Assert.fail();
+        }
+    }
+
+    @Test(groups = "certificate", priority = 2, dependsOnMethods = "testAddApplicationWithCertificate")
+    public void testGetApplicationWithServerErrorWhenRetrievingCert() throws CertificateMgtException {
+
+        reset(applicationCertificateManagementService);
+        doThrow(serverException).when(applicationCertificateManagementService).getCertificate(anyInt(), anyString());
+        try {
+            applicationManagementService.getApplicationByResourceId(appIdWithCert, SUPER_TENANT_DOMAIN_NAME);
+            Assert.fail();
```

Comment:
Add a comment mentioning successful retrieval of the app without an exception means the faillure of the test case.